File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> map = new HashMap<>();
        String answer = "";
        
        for(String name : participant) {
            map.put(name, map.getOrDefault(name, 0) + 1);
        }
        
        for(String name : completion) {
            map.put(name, map.get(name) - 1);
        }
        
        for(String name : participant) {
            if(map.get(name) != 0) answer = name;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import Counter

def solution(participant, completion):
    # dic = Counter(participant)
    
    dic = {}
    for p in participant:
        if p not in dic:
            dic[p] = 1
        else :
            dic[p] += 1
        
    for c in completion:
        dic[c] -= 1
    
    for key in dic.keys():
        if dic[key] > 0:
            return key
        
    return ""